modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -179,16 +186,25 @@ public AsyncIndexBySearchAction(BulkByScrollTask task, ESLogger logger, ParentTa
         @Override
         protected ScrollableHitSource buildScrollableResultSource(BackoffPolicy backoffPolicy) {
             if (mainRequest.getRemoteInfo() != null) {
-                // NORELEASE track 500-level retries that are builtin to the client
                 RemoteInfo remoteInfo = mainRequest.getRemoteInfo();
+                HttpClientBuilder httpClient = HttpClients.custom();
                 if (remoteInfo.getUsername() != null) {
-                    // NORELEASE support auth
-                    throw new UnsupportedOperationException("Auth is unsupported");
+                    UsernamePasswordCredentials creds = new UsernamePasswordCredentials(remoteInfo.getUsername(), remoteInfo.getPassword());
```

Comment:
There are lots of plain text passwords in this commit - like in memory and buffers and stuff. We *allow* you to send the password plain text if you set the host as `http://otherhost:9200` rather than `https://otherhost:9200` but we don't force it. Which plain text password do you mean?